src/main/java/com/feedzai/commons/sql/abstraction/batch/AbstractBatch.java

```diff
@@ -302,7 +302,7 @@ public void flush(boolean sync) {
     }
 
     @Override
-    public synchronized void run() {
+    public void run() {
         if (System.currentTimeMillis() - lastFlush >= batchTimeout) {
```

Comment:
This lastFlush variable should be made volatile given the removal of the synchronized. Otherwise there's no guarantee that this thread will see it updated (if the flush call is performed from an external thread via the public api of the batch, or if the executor had more threads --- which is not the case currently).